xilem_web/web_examples/fetch/src/main.rs

```diff
@@ -0,0 +1,181 @@
+        .send()
+        .await?
+        .json::<Vec<Cat>>()
+        .await?
+        .into_iter()
+        .take(count)
+        .collect())
+}
+
+struct AppState {
+    cats_to_fetch: usize,
+    cats_are_being_fetched: bool,
+    cats: Vec<Cat>,
+    debounce_in_ms: usize,
+    reset_debounce_on_update: bool,
+    error: Option<String>,
+}
+
+impl Default for AppState {
+    fn default() -> Self {
+        Self {
+            cats_to_fetch: 0,
+            cats: Vec::new(),
+            debounce_in_ms: 1000,
+            cats_are_being_fetched: false,
+            reset_debounce_on_update: true,
+            error: None,
+        }
+    }
+}
+
+pub fn input_target<T>(event: &T) -> web_sys::HtmlInputElement
+where
+    T: JsCast,
+{
+    event
+        .unchecked_ref::<web_sys::Event>()
+        .target()
+        .unwrap_throw()
+        .unchecked_into::<web_sys::HtmlInputElement>()
+}
+
+fn app_logic(state: &mut AppState) -> impl HtmlDivElement<AppState> {
+    let cat_images = state
+        .cats
+        .iter()
+        .map(|cat| img(()).src(cat.url.clone()))
+        .collect::<Vec<_>>();
+    div((
+        fieldset((
+            legend("Cat fetch controls"),
+            table((
+                tr((
+                    td(
+                        label("Reset fetch debounce timeout when updating the cat count:")
+                            .for_("reset-debounce-update"),
+                    ),
+                    td(input(())
+                        .id("reset-debounce-update")
+                        .attr("type", "checkbox")
+                        .attr("checked", state.reset_debounce_on_update)
+                        .on_input(|state: &mut AppState, event: web_sys::Event| {
+                            state.reset_debounce_on_update = input_target(&event).checked();
+                        })),
+                )),
+                tr((
+                    td(label("Debounce timeout in ms:").for_("debounce-timeout-duration")),
+                    td(input(())
+                        .id("debounce-timeout-duration")
+                        .attr("type", "number")
+                        .attr("min", 0)
+                        .attr("value", state.debounce_in_ms)
+                        .on_input(|state: &mut AppState, ev: web_sys::Event| {
+                            state.debounce_in_ms = input_target(&ev).value().parse().unwrap_or(0);
+                        })),
+                )),
+                tr((
+                    td(label("How many cats would you like?").for_("cat-count")),
+                    td(input(())
+                        .id("cat-count")
+                        .attr("type", "number")
+                        .attr("min", 0)
+                        .attr("value", state.cats_to_fetch)
+                        .on_input(|state: &mut AppState, ev: web_sys::Event| {
+                            if !state.cats_are_being_fetched {
+                                state.cats.clear();
```

Comment:
Yeah you're right, though I rather wanted to have a clear visual state when images are fetched (i.e. I consider that a feature not a bug^^), to see when a new request is started, so that they don't slowly pop in over the previous images, when they're being loaded, I think the "weirdness" will be further amplified, when setting width/height on each.

Ideally, I think the `fetch_cats` prefetches the images, and when it's resolved all images are changed at once in one frame. But that requires quite a bit [more code](https://users.rust-lang.org/t/a-future-for-loading-images-via-web-sys/42370/2) (I actually tried just something like the equivalent of js: `Promise.all([fetch(), fetch()])` but that didn't seem to be enough, as the images were redownloaded, probably different headers or something like that).